Library/Homebrew/test/support/helper/test_case.rb

```diff
@@ -13,6 +13,16 @@ class TestCase < ::Minitest::Test
     TEST_SHA256 = "deadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeefdeadbeef".freeze
 
     def setup
+      [
+        HOMEBREW_LIBRARY/"Taps/homebrew/homebrew-core/Formula",
+        HOMEBREW_CACHE,
+        HOMEBREW_CACHE_FORMULA,
+        HOMEBREW_LOCK_DIR,
+        HOMEBREW_CELLAR,
+        HOMEBREW_LOGS,
+        HOMEBREW_TEMP,
+      ].each(&:mkpath)
```

Comment:
Might be worth having a comment explaining that this needs to be done before `super` because of `FSLeakLogger`?